src/ImageSharp/Formats/ImageDecoderUtilities.cs

```diff
@@ -14,42 +14,21 @@ internal static class ImageDecoderUtilities
         /// Reads the raw image information from the specified stream.
         /// </summary>
         /// <param name="decoder">The decoder.</param>
-        /// <param name="stream">The <see cref="Stream"/> containing image data.</param>
-        public static async Task<IImageInfo> IdentifyAsync(this IImageDecoderInternals decoder, Stream stream)
-        {
-            if (stream.CanSeek)
-            {
-                return decoder.Identify(stream);
-            }
-
-            using MemoryStream ms = decoder.Configuration.MemoryAllocator.AllocateFixedCapacityMemoryStream(stream.Length);
-            await stream.CopyToAsync(ms).ConfigureAwait(false);
-            ms.Position = 0;
-            return decoder.Identify(ms);
-        }
+        /// <param name="stream">The <see cref="BufferedReadStream"/> containing image data.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="stream"/> is null.</exception>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public static Task<IImageInfo> IdentifyAsync(this IImageDecoderInternals decoder, BufferedReadStream stream)
+            => Task.FromResult(decoder.Identify(stream));
```

Comment:
In case of error we should return `Task.FromException` instead of failing to create the `Task`.

The fix is on the way, as I'm working on a PR for #1280, so don't bother with it.

Reply:
Sweet. Will wait to see what you come up with then. 👍